### ISSUE TITLE
reexport AD.ForwardDiffBackend()

### ISSUE THREAD

**Issue by thorek1** [gdalle/ImplicitDifferentiation.jl#116]
in order to keep dependencies lean in downstream packages I would recommend to reexport ForwardDiffBackend()
```Julia
import Reexport
Reexport.@reexport import AbstractDifferentiation: ForwardDiffBackend
```

**Comment by gdalle** [comment]
I disagree, it doesn't change anything to leanness (at least in the manifest) and it's good to make users aware that conditions backends are provided by a third party

**Comment by gdalle** [comment]
Partly because the conditions backends have bugs that we're not responsible for 😅

**Comment by thorek1** [comment]
let me rephrase from a user perspective:
importing further packages (extensions are a different case) in order to make use of certain options within the package adds complexity which from the user perspective is not obvious as to why it is so.

reexporting it also takes the compat burden of the fellow package maintainers downstream :)

**Comment by gdalle** [comment]
I think the user who deems it necessary to switch conditions backends is advanced enough to not be afraid of importing AbstractDifferentiation. Besides, the fact that AbstractDifferentiation exports its backends is a relic of times past: in the not so distant future, it will hopefully be the backends themselves who define these objects in package extensions
